src/mapml-viewer.js

```diff
@@ -17,20 +18,18 @@ export class MapViewer extends HTMLElement {
   }
   set controls(value) {
     const hasControls = Boolean(value);
-    if (hasControls)
-      this.setAttribute('controls', '');
-    else
+    if (hasControls) {
+      this.setAttribute('controls','');
+    } else {
       this.removeAttribute('controls');
-    this._toggleControls(hasControls);
+    }
   }
-  get controlslist() {
-    return this.hasAttribute('controlslist') ? this.getAttribute("controlslist") : "";
+  get controlsList() {
+    return this._controlsList;
   }
-  set controlslist(val) {
-    let options = ["nofullscreen", "nozoom", "nolayer", "noreload"],
-        lowerVal = val.toLowerCase();
-    if (this.controlslist.includes(lowerVal) || !options.includes(lowerVal))return;
-    this.setAttribute("controlslist", this.controlslist+` ${lowerVal}`);
+  set controlsList(value) {
```

Comment:
I noticed that if you have got a map state set with, for example conrolslist="nofullscreen" or anything else., if you do a map.removeAttribute("controlslist") it doesn't change the state of the map. You have to do a map.controlsList = "" to achieve that effect.  Not sure it's a big deal; probably an artifact of having the getter / setter property name be different than the attribute name.

Reply:
This is recently fixed, good work!